### ISSUE TITLE
iOS - Auth Login WebAuthenticator crash with Universal linking

### ISSUE THREAD

**Issue by usmaanirfan** [dotnet/maui#21183]
### Description

Software engineer

### Steps to Reproduce

1. Create an instance of ex:  authResult = await WebAuthenticator.Default.AuthenticateAsync(
                new WebAuthenticatorOptions()
                {
                    Url = startUri,
                    CallbackUrl = new Uri("........./callback"),
                    PrefersEphemeralWebBrowserSession = true
                });

2. Add universal linking domain in entitlements

4. After login click on the auth webpage it crash App while trying to callback and navigate back in the App with code.

### Version with bug

8.0.6 SR1

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 17

### Did you find any workaround?

No

### Relevant log output

```shell
=================================================================
	Native Crash Reporting
=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Native stacktrace:
=================================================================
	0x1048c25b4 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x1048ad4a0 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x10488f828 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x217a68888 - /usr/lib/system/libsystem_platform.dylib : <redacted>
	0x1ce666210 - /System/Library/Frameworks/SafariServices.framework/SafariServices : <redacted>
	0x1afd7087c - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1b031b0b4 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afd703a0 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afee3620 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afee32d8 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afccbfe0 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : __UIVIEW_IS_EXECUTING_ANIMATION_COMPLETION_BLOCK__
	0x1afccb778 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afccadf0 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afcad00c - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afcad120 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1af097680 - /System/Library/Frameworks/QuartzCore.framework/QuartzCore : <redacted>
	0x1b5a16300 - /usr/lib/system/libdispatch.dylib : <redacted>
	0x1b5a24998 - /usr/lib/system/libdispatch.dylib : <redacted>
	0x1b5a245b0 - /usr/lib/system/libdispatch.dylib : _dispatch_main_queue_callback_4CF
	0x1ada50f9c - /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : <redacted>
	0x1ada4dca8 - /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : <redacted>
	0x1ada4d3f8 - /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : CFRunLoopRunSpecific
	0x1f0fdb4f8 - /System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices : GSEventRunModal
	0x1afe738a0 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afe72edc - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : UIApplicationMain
	0x1045805f4 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : xamarin_UIApplicationMain
	0x1048d2e3c - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x1048d173c - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x1048c6148 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x1048c3d14 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x1048929a0 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x10483a16c - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x104840468 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x104897f98 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x1045acfb8 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : xamarin_get_original_working_directory_path
	0x104900504 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x1d07a2dcc - /usr/lib/dyld : <redacted>

=================================================================
	Basic Fault Address Reporting
=================================================================
Memory around native instruction pointer (0x1a5d8f304):0x1a5d8f2f4  21 38 2e 91 42 05 00 14 00 00 00 ea 6d ff ff 54  !8..B.......m..T
0x1a5d8f304  10 00 40 f9 02 82 7d 92 51 10 40 f9 31 03 10 36  ..@...}.Q.@.1..6
0x1a5d8f314  f0 03 00 36 11 fe 77 d3 91 fe ff b4 3f 06 00 f1  ...6..w.....?...
0x1a5d8f324  60 02 00 54 11 20 e0 d2 11 02 11 cb e1 03 10 aa  `..T. ..........

=================================================================
	Managed Stacktrace:
=================================================================
	  at <unknown> <0xffffffff>
	  at UIKit.UIApplication:xamarin_UIApplicationMain <0x000b8>
	  at UIKit.UIApplication:UIApplicationMain <0x0006e>
	  at UIKit.UIApplication:Main <0x0015a>
	  at MauiAppHybrid.Program:Main <0x0002c>
	  at <Module>:runtime_invoke_direct_void_string[] <0x0007e>
	  at <unknown> <0x00000>
=================================================================

=================================================================
	Native Crash Reporting
=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

An error has occurred in the native fault reporting. Some diagnostic information will be unavailable.

=================================================================
	Native stacktrace:
=================================================================
	0x1048c25b4 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x1048ad4a0 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x10488f828 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x217a68888 - /usr/lib/system/libsystem_platform.dylib : <redacted>
	0x1ce666210 - /System/Library/Frameworks/SafariServices.framework/SafariServices : <redacted>
	0x1afd7087c - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1b031b0b4 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afd703a0 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afee3620 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afee32d8 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afccbfe0 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : __UIVIEW_IS_EXECUTING_ANIMATION_COMPLETION_BLOCK__
	0x1afccb778 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afccadf0 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afcad00c - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afcad120 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1af097680 - /System/Library/Frameworks/QuartzCore.framework/QuartzCore : <redacted>
	0x1b5a16300 - /usr/lib/system/libdispatch.dylib : <redacted>
	0x1b5a24998 - /usr/lib/system/libdispatch.dylib : <redacted>
	0x1b5a245b0 - /usr/lib/system/libdispatch.dylib : _dispatch_main_queue_callback_4CF
	0x1ada50f9c - /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : <redacted>
	0x1ada4dca8 - /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : <redacted>
	0x1ada4d3f8 - /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : CFRunLoopRunSpecific
	0x1f0fdb4f8 - /System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices : GSEventRunModal
	0x1afe738a0 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x1afe72edc - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : UIApplicationMain
	0x1045805f4 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : xamarin_UIApplicationMain
	0x1048d2e3c - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x1048d173c - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x1048c6148 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x1048c3d14 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x1048929a0 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x10483a16c - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x104840468 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x104897f98 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x1045acfb8 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : xamarin_get_original_working_directory_path
	0x104900504 - /private/var/containers/Bundle/Application/3AABA233-B5AC-49CD-8865-68C027AEE3BF/MauiAppHybrid.app/MauiAppHybrid : AppleCryptoNative_X509ImportCollection
	0x1d07a2dcc - /usr/lib/dyld : <redacted>

Exiting early due to double fault.
```

**Comment by usmaanirfan** [comment]
Hi, I am still able to reproduce this issue. Do you want me to share my source code for this? If Yes could you please let me know where I can share that privately?

**Comment by jfversluis** [comment]
> Do you want me to share my source code for this? If Yes could you please let me know where I can share that privately?

We would like to get a reproduction, which doesn't mean your actual project, especially if that is something that is private. Unfortunately, we can't look at closed-source code.

A reproduction means that you lift the relevant pieces that lead up to this behavior from your project and into a new project that only has the important code so we can easily see what might be going on and at the same time might help you understand better all the code that is involved. It wouldn't be the first time that people figure out the solution by creating the reproduction.

Also see: https://github.com/dotnet/maui/blob/main/.github/repro.md

I understand that its a bit harder here since there is a server-side involved. It would be great if you could create the project, see the bug and then provide the code to us.

**Comment by usmaanirfan** [comment]
Hi @jfversluis ,

Here is the generic repo for the https://github.com/usmaanirfan/MauiAppUniversalLinking

There are blank urls and test bundle and domains because of privacy. I hope the source code can give you high level idea.

When I run this on my iOS device and login it tries to redirect back to the Ann from browser with code and then crash.

**Comment by zishani** [comment]
Same issue here.

For me it woks on the simulator for iPhone (iOS) but crashes on the simulator for iPad (iPadOS).

**Comment by QianaJiao** [comment]
Verified this issue with Visual Studio 17.10.0 Preview 5 (8.0.21& 8.0.7). I can repro this issue on iOS 17.2 with https://github.com/usmaanirfan/MauiAppUniversalLinking

**Comment by zizusoft** [comment]
> Verified this issue with Visual Studio 17.10.0 Preview 5 (8.0.21& 8.0.7). I can repro this issue on iOS 17.2 with https://github.com/usmaanirfan/MauiAppUniversalLinking
> 
> 

Are you able to confirm the issue is only with iPadOS (iPad) and not iOS (iPhone) ?

Thanks

**Comment by zizusoft** [comment]
Managed to resolve this by compiling against Xcode 15.2

**Comment by usmaanirfan** [comment]
Hi, I have shared repo in my previous comment and reproduction steps. Should there be anything? I would be happy to share.

**Comment by mattleibow** [comment]
> Managed to resolve this by compiling against Xcode 15.2

Does this help @usmaanirfan ?

**Comment by Marini88** [comment]
I'm getting this on Xcode 16.1       🙌

**Comment by IVSoftware** [comment]
For some reason, this thread hadn't turned up on my search, but I have closed #26814 as clearly a duplicate.

I had also posted on SO if any of the details might be useful https://stackoverflow.com/q/79306670/5438626..

**XCode is 16.2**

~~~plaintext
> dotnet workload list

Installed Workload Id      Manifest Version       Installation Source
---------------------------------------------------------------------
android                    35.0.7/9.0.100         VS 17.12.35527.113
ios                        18.1.9163/9.0.100      VS 17.12.35527.113
maccatalyst                18.1.9163/9.0.100      VS 17.12.35527.113
maui-windows               9.0.0/9.0.100          VS 17.12.35527.113
~~~
